internal/cliutils/versionutil.go

```diff
@@ -15,7 +15,10 @@ func CheckPackageOperatorVersion(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	if operatorVersion[1:] != config.Version {
+	if config.Version == "dev" && operatorVersion != "" {
+		fmt.Fprintf(os.Stderr, "❗ Glasskube CLI version is dev but the operator version is %s\n", operatorVersion[1:])
+	}
+	if operatorVersion[1:] != config.Version && config.Version != "dev" {
```

Comment:
The additional dev build check here could be removed if you use an `else if` instead.